### PR TITLE
NF: Choose backend

### PR DESCRIPTION
**Pull request by bpoldrack**
Commits from earlier, not yet merged changes included. 
Topic is about last commits starting with: 
https://github.com/datalad/datalad/commit/85db7423db0c75a5afcb19c6b31ac76325e08f1b

### CODE REVIEW

File: datalad/support/annexrepo.py [hunk 1]
```diff
@@ -410,11 +422,12 @@ def annex_addurl_to_file(self, file, url, options=[]):
 
         annex_options = ['--file=%s' % file[0]] + options + [url]
         self._run_annex_command('addurl', annex_options=annex_options,
-                                log_online=True, log_stderr=False)
+                                backend=None, log_online=True,
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: shouldn't it use provided backend?

File: datalad/support/annexrepo.py [hunk 2]
```diff
@@ -426,7 +439,8 @@ def annex_addurls(self, urls, options=[]):
         """
 
         self._run_annex_command('addurl', annex_options=options + urls,
-                                log_online=True, log_stderr=False)
+                                backend=None, log_online=True,
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: here as well --pass the backend

Comment (yarikoptic) [inline datalad/support/annexrepo.py]: which all suggests that ideally there should be a unittest ... and `get_file_backend` function?

File: datalad/support/annexrepo.py [hunk 3]
```diff
@@ -499,3 +513,29 @@ def annex_whereis(self, files):
...
+            files to migrate.
+        backend: str
+            specify the backend to migrate to. If none is given, the
+            default backend of this instance will be used.
+        """
+
+        if self.is_direct_mode():
+            raise CommandNotAvailableError(
+                'git-annex migrate',
+                "Command 'migrate' is not available in direct mode.")
```
Comment (yarikoptic) [inline datalad/support/annexrepo.py]: oh -- I didn't know ;)

File: datalad/support/annexrepo.py (hunk 1 above)
Comment (bpoldrack) [inline datalad/support/annexrepo.py]: Of course!

File: datalad/support/annexrepo.py (hunk 3 above)
Comment (bpoldrack) [inline datalad/support/annexrepo.py]: Neither did I :)
But doesn't matter, the main reason we need for leads to migrating before the repo is cloned to a windows system anyway. And use MD5 as default when creating handles on windows, of course.

File: datalad/support/annexrepo.py (hunk 2 above)
Comment (bpoldrack) [inline datalad/support/annexrepo.py]: Will do.

**Pull request merged**